src/components/Single-Article.jsx

```diff
@@ -25,12 +28,47 @@ export const SingleArticle = () => {
             setComments(body.comments)
         })
     }, [article_id])
+
+    useEffect(() => {
+        getUsers()
+        .then((body) => {
+            setUsers(body.users)
+    })
+    }, [])
     
     const handleVote = (article_id, vote) => {
             patchVotes(article_id, vote);
             setVoteChanges(prev => ({...prev, [article_id]: vote}));
             }
 
+    const handleUsername = (event) => {
+        setComment({ ...comment, username: event.target.value})
```

Comment:
use a callback function here to access the latest value of the comment and update that rather than directly using the state - this will help you manage the asynchronicity  of the set state function

Reply:
See this: https://www.dhiwise.com/post/guide-to-state-management-with-react-setstate-callback